chapters/2.md

```diff
@@ -1,211 +1,170 @@
-
-*One*.
-
-And for events?
-
-![global state + events](/images/chapters/2/09.png)
-
-*One*. Each different little component can only dispatch an event to the store. And what makes the store so special?
-
-It reduces the amount of things a programmer has to think about from potentially everything, to a really simple formula.
+```html
+  <dom-module id="simple-element">
+    ...beautiful element stuff...
+  </dom-module>
+```
 
+  While this element sounds like something out of a Star Wars episode, it's not.
+Although it's still pretty awesome! Get ready for this incredible definition...
+the `<dom-module>` element is used to specify a module of DOM. :-P If that doesn't
+make sense, fear not. I'll go into detail.
+
+### The DOM
+HTML consists of many different elements which come in the form
+  ```html
+    <element-name>
+      ...some content...
+    </element-name>
+  ```
+where an element has opening and closing tags and some content in between. For
+example, in `simple-element.html` we can see the following:
+```html
+  <h2>[[prop1]]!</h2>
 ```
-new state = old state + event
+In this example, the opening and closing `<h2>` tags surround some content. So,
+how is it that the browser understands what this snippet means and where to place
+the content? The browser understands by constructing an model that represents the
+document called the *Document Object Model*, or the *DOM*. This is really helpful
+for us because it doesn't have to be static. For example, let's say the browser receives
+an HTML file and constructs the DOM associated with that page. In order to update
+the web page we can just update the DOM in the browser. This means content can be
+dynamically added and removed from a given web page at runtime.
+
+TODO Refine that above. Not well worded.
+
+### The Module  
+So, that's all great. But what's meant by *module* here? Have you every heard the
+phrase "Don't Repeat Yourself?" This phrase is commonly smashed into the acronym
+DRY. DRY is a quick and easy way of suggesting that software should be designed
+in a manner that maximizes reuse of code which, in turn, maximizes the value
+offered by that code and minimizes the number of changes required to modify that
+code. This isn't the only benefit afforded by modular code but it's a big one!
+It's in this vein that we use the term module. A module is a collection of related
+code.
+
+### So, What's a DOM Module?
+Returning to our discussion of the `<dom-module>` element above, the ridiculous
+definition I used is starting to make sense, right? The `<dom-module>` element
+is, in fact, used to define a module of DOM. That is, a collection of related
+code that defines the model the browser sees when our custom element is seen in
+an HTML file.
+
+TODO should I refine the DOM MODULE lesson above? Good but could come together
```

Comment:
The breakdown of the name is great (dom, then module). I think you could refine the module portion by emphasizing the problem it's introduced to solve. Definitely keep the reusability stuff, and then go more in on: ok, so we have some code we want to reuse. How do we do that? Well, we need to be able to export/import it so we can reuse it all over the place! And that's why dom-module is important. So in general the most important thing `dom-module` does is give us a way to get a handle on the process: providing a way to import/export, as well as, provide a way to get a handle on it via id.

Reply:
I really don't like the idea of introducing the problem that dom modules are designed to solve at this point in the lesson. That could come in further sections. The reason I say this is because our goal in this is to get people up and running with polymer as fast as possible and there are some chief constraints with that:
1. Many devs that we hire have either very little experience or only have experience with one framework. Introducing ideas such as how the dom module solves the reusability issue is a great idea but opens a rabbit hole that'll only confuse the scoped ideas presented. My pure goal is to say, "this is a simple web component and it's running on your machine! That wasn't so bad, was it?"
2. Design is important but syntax and application is paramount. By this I mean that describing how polymer is used will provide more immediate value than describing the design decisions made during the creation of polymer (or web components) itself. So, while you and I would love to know that information I think it only obscures the lesson at this point in time for the user.
All the user needs to know is that they need to import polymer (pretty straightforward idea; without importing it presumably it won't be available), there are modules with allow definition of markup and functionality, there is this thing called a template which identifies a blueprint (or, as I say in the writing, a drawing template) of a custom element, and finally that I can define a custom element recognized by the browser. Anything outside of that I want to avoid because it'll increase the cognitive load and will provide little benefit at this point in the devs learning process.

Reply:
I like mentioning that the dom-module aspect is what allows the code to be reusable. Probably any developer that comes here has used basic html and knows that you can't replicate it like calling a function, unless you copy and paste. So the dom-module provides something new and helpful. But keep that explanation as simple and brief as possible.